app/jobs/benchmark_regression_job.rb

```diff
@@ -0,0 +1,72 @@
+require 'uri'
+require 'net/http'
+require 'json'
+
+class BenchmarkRegressionJob < ActiveJob::Base
+  queue_as :default
+
+  @@highest = Hash.new(10000) # any max value
+  @@flag = Hash.new(false)
+
+  def perform(benchmark_run_id)
+
+    benchmark_run = BenchmarkRun.find_by_id(benchmark_run_id)
+
+    categories = benchmark_run.result.keys
+
+    current_commit = benchmark_run.initiator
+    previous_commits = Commit.where('created_at < ?', current_commit.created_at).where(repo: current_commit.repo).limit(1000)
+
+    previous_benchmark_runs = []
+    previous_commits.each do |commit|
+      previous_benchmark_runs << BenchmarkRun.find_by(initiator: commit, benchmark_type: benchmark_run.benchmark_type,
```

Comment:
Not sure how I missed this previously but it is going to hit the database `previous_commits` times